core/src/main/java/org/apache/calcite/rel/logical/LogicalProject.java

```diff
@@ -137,14 +170,20 @@ public static LogicalProject create(final RelNode input, List<RelHint> hints,
 
   @Override public RelNode withHints(List<RelHint> hintList) {
     return new LogicalProject(getCluster(), traitSet, hintList,
-        input, getProjects(), getRowType());
+        input, getProjects(), getRowType(), variablesSet);
+  }
+
+  @Override public RelWriter explainTerms(RelWriter pw) {
+    return super.explainTerms(pw)
+        .itemIf("variablesSet", variablesSet, !variablesSet.isEmpty());
```

Comment:
Please print `variablesSet` **before** the regular project fields as it would be easier to relate project and its variables.

Reply:
fixed